R/dash.R

```diff
@@ -1056,6 +1087,41 @@ Dash <- R6::R6Class(
       self$config$silence_routes_logging <- dev_tools_silence_routes_logging
       self$config$props_check <- dev_tools_props_check
 
+      if (private$debug && self$config$ui) {
+        self$server$on('before-request', function(server, ...) {
+          self$server$set_data("timing_information", list(
+            "__dash_server" = list(
+              "dur" = as.numeric(Sys.time()),
+              "desc" = NULL
+            )
+          ))
+        })
+
+        self$server$on('request', function(server, request, ...) {
```

Comment:
We will use `request` rather than `after-request`, which sounds desirable given our use of `after_request` in Dash for Python, because `after-request` in Fiery

> _Will trigger after the request event. This can be used to **inspect the response (but not modify it) before it is send to the client**. The handlers will receive the app itself as the server argument, the client id as the id argument, the request object as the request argument, and the response as the response argument. Any return value is discarded._

Compare with the behaviour of Fiery's `request`:

> _Will trigger after the before-request event. This is where the main request handling is done. **The return value of the last handler is send back to the client as response.** If no handler is registered a 404 error is returned automatically. If the return value is not a valid response, a 500 server error is returned instead. The handlers will receive the app itself as the server argument, the client id as the id argument, the request object as the request argument, and the list of values created by the before-event handlers as the arg_list argument._